rust/protocol/src/error.rs

```diff
@@ -1,68 +1,182 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
-use crate::curve::KeyType;
+//! Errors that may occur during various stages of the Signal Protocol.
 
+#![warn(missing_docs)]
+
+use crate::curve::{AsymmetricRole, KeyType};
+
+#[cfg(doc)]
+pub use crate::{
+    address::ProtocolAddress,
+    curve::{PrivateKey, PublicKey},
+    group_cipher::{group_decrypt, group_encrypt},
+    protocol::{SenderKeyMessage, SignalMessage, CIPHERTEXT_MESSAGE_CURRENT_VERSION},
+    ratchet::{ChainKey, RootKey},
+    sealed_sender::{sealed_sender_decrypt, sealed_sender_multi_recipient_encrypt},
+    sender_keys::SenderKeyRecord,
+    session_cipher::{message_decrypt, message_encrypt},
+    state::{PreKeyId, PreKeyRecord, SignedPreKeyId, SignedPreKeyRecord},
+    storage::{IdentityKeyStore, PreKeyStore, SenderKeyStore, SessionStore, SignedPreKeyStore},
+};
+#[cfg(doc)]
+use uuid::Uuid;
```

Comment:
Let's put these below the regular imports so that someone scanning imports doesn't get caught on these.